server/db/order_product_join.js

```diff
@@ -0,0 +1,10 @@
+const Sequelize = require ('sequelize');
+const db = require('./_db');
+
+const Order_Product = db.define('order_product', {
+  quantity: {
+    type: Sequelize.INTEGER
+  }
+});
+
+module.exports = Order_Product;
```

Comment:
ISSUE: we do not retain the information of the product at the time of purchase, we only reference to its current state so order details could be incorrect

Reply:
we can: 
1) back end- already done, but have this on the join table
2) front-end - reassign quantity in order/product obj, as order quantity rather than stock?

Reply:
Notes post-dicussion: we want to create a hook that updates a "purchase price" field in the join table only after the order status changes to "complete"

Reply:
We got way ahead of ourselves on this one. Going to paste some code and notes below for some progress that we made on the hook/ options that might end up less hairy. There are a few options for when to set/how to decide the purchase price when we get to it, but now is not the time to deal with this priority-wise because we might not even get to being able to update our cart at this point.

Reply:
Im grumpy

Reply:
{
  hooks: {
    afterUpdate: (product, options) => {
      if(product.changed('price')) {
        Order_Product.findAll({
          where: {productId: product.id},
          include: [{
            model: Order,
            where: {status: 'pending'}
          }]
        })
        .then(orders => {
          console.log('orders', orders)
        })
      }
     
    }
  }
}

We may also do this: purchase price in order_product will be null by default.  When an item is added to a cart and/or when a cart is accessed, the price will be queried from the Product table.  After checkout is completed and the order is set to completed status, then you will set purchaseprice.